File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

public class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int[] rank = new int[k];
        int size=0;

        for (int i = 0; i < score.length; i++) {
            if (size<k) {
                rank[size]=score[i];
                size++;
                Arrays.sort(rank, 0, size);
            }
            else {
                if (score[i]>rank[0]) {
                    // 점수 업데이트
                    rank[0]=score[i];
                    Arrays.sort(rank);
                }
            }
            answer[i]=rank[0];
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    let stack = []
    
    for(const s of score) {
        stack.push(s)
        stack.sort((a, b)=>b-a)
        if(stack.length>k) {
            stack.pop();
        }
        answer.push(stack.at(-1))
    }
    return answer;
}